code/control/DNRoot.php

```diff
@@ -2438,4 +2440,93 @@ public function doCreateProject($data, $form) {
 		return $this->redirectBack();
 	}
 
+	/**
+	 * Returns the state of a current project build.
+	 *
+	 * @param SS_HTTPRequest $request
+	 *
+	 * @return SS_HTTPResponse
+	 */
+	public function createprojectprogress(SS_HTTPRequest $request) {
+		$project = $this->getCurrentProject();
+		if(!$project) {
+			return $this->httpError(404);
+		}
+
+		if(!$project->canView()) {
```

Comment:
canView isn't needed as `getCurrentProject()` goes through the process of doing that already via `DNProjectList()`